### PR TITLE
Add JSON output to version command

### PR DESCRIPTION
**Pull request by karolz-ms**

### CODE REVIEW

File: cli/azd/cmd/version.go [hunk 1]
```diff
@@ -10,20 +10,42 @@ import (
...
+	formatter, err := output.GetFormatter(cmd)
+	if err != nil {
+		return err
+	}
+
+	switch formatter.Kind() {
+	case output.NoneFormat:
+		fmt.Printf("azd version %s\n", internal.Version)
+	case output.JsonFormat:
+		versionSpec := internal.GetVersionSpec()
```
Comment (ellismg) [inline cli/azd/cmd/version.go:43]: There was some discussion at one point about extending `version` to print version numbers of other tools we depend on.

Perhaps for safety right now we should instead format this as an object that looks like:

```
{
  "azd": "<version-number>"
}
```

So if we go add more versions later, we don't break consumers?

Comment (karolz-ms) [reply cli/azd/cmd/version.go:43]: @ellismg I like that. How about combining your and mine idea and having something like this

```json
{ 
  "azd": {
    "version": "1.2.3",
     "commit": "12345abc"
  }
}
```

Comment (karolz-ms) [reply cli/azd/cmd/version.go:43]: Pushed a related change--LMK what you think!

**Review by ellismg — APPROVED** [review]
LGTM

File: cli/azd/cmd/version.go (hunk 1 above)
Comment (ellismg) [reply cli/azd/cmd/version.go:43]: Love it!

**Pull request merged**